dbt/adapters/redshift/connections.py

```diff
@@ -38,33 +42,156 @@ class RedshiftConnectionMethod(StrEnum):
 
 
 @dataclass
-class RedshiftCredentials(PostgresCredentials):
+class RedshiftCredentials(Credentials):
+    host: str
+    user: str
+    port: Port
     method: str = RedshiftConnectionMethod.DATABASE  # type: ignore
     password: Optional[str] = None  # type: ignore
     cluster_id: Optional[str] = field(
         default=None,
         metadata={"description": "If using IAM auth, the name of the cluster"},
     )
     iam_profile: Optional[str] = None
-    iam_duration_seconds: int = 900
-    search_path: Optional[str] = None
-    keepalives_idle: int = 4
     autocreate: bool = False
     db_groups: List[str] = field(default_factory=list)
     ra3_node: Optional[bool] = False
+    connect_timeout: int = 30
+    role: Optional[str] = None
+    sslmode: Optional[str] = None
+    application_name: Optional[str] = "dbt"
```

Comment:
@sathiish-kumar sorry I didn't spot this sooner, we actually don't want this to be configurable in dbt-redshift (though it is in dbt-postgres)